daget/exceptions.py

```diff
@@ -1,5 +1,14 @@
 class ResolveError(ValueError):
-    pass
+    def __init__(self, message, http_response_code=None):
+        super().__init__(message)
+        self.http_response_code = http_response_code
 
-class RepoError(Exception):
-    pass
\ No newline at end of file
+class RepoError(ValueError):
+    def __init__(self, message, url, supported_urls=None, http_response_code=None):
+        super().__init__(message)
+        self.url = url
+        self.supported_urls = supported_urls or ["dataverse.harvard.edu", "dataverse.no", "snd.se/catalogue", "su.figshare.com", "figshare.scilifelab.se", "zenodo.org"]
```

Comment:
hardcoded list of repository url:s should be removed.
daget should try to get a file list via schema.org distribution (if it´s not figshare or zenodo) and if this fails it should throw the error instead. keeping a list of all suported url:s in the source coude is not a sustainable soultion